woocommerce/woocommerce-ios
# [Orders with Coupons M6] Remove product from order on collapsable product row component

Closes: #10834 
#10844  and #10845 need to be merged first

## Description
This PR adds behaviour to the existing `Remove product from order` button to the collapsible product row component, by adding a callback closure which triggers the existing logic that was used previously to remove products from orders.

We add a wrapper around this method for convenience, so we can just pass the `ProductRowViewModel` unique ID rather than the `OrderItem` object.

![Remove Product](https://github.com/woocommerce/woocommerce-ios/assets/3812076/a888e0db-15a6-4b80-81f6-06c723671028)

## Testing instructions
- Switch `ordersWithCouponsM6` feature flag to `true`
- Go to `Orders` > `+` > `+ Add Products` > Add one or more products to your order.
- Tap on the chevron icon of a product card in order to expand the card, then tap `Remove product from order`
- Confirm how the product is removed from the order we're currently editing, as well as the created order after tapping `create`
